payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -281,6 +288,14 @@ internal class PaymentSheetViewModel internal constructor(
         }
     }
 
+    suspend fun confirmStripeIntent(confirmStripeIntentParams: ConfirmStripeIntentParams) {
+        paymentController.startConfirmAndAuth(
```

Comment:
hmm now with authActivityStarterHost passed from `PaymentSheetActivity`, we'll need to pass the additional param to `checkout` and not always use it. Having a separate interface seems clearer here

Reply:
When running through the code it is always confusing because you have to bounce (via startConfirm) up to the activity and then back to the viewModel again just to get the activity.   

If you pass the activity to the viewModel (and construct the authStarterHost) then the activity I think is used in all cases, it would also simplify a similar problem with the googleLauncher.

Reply:
as discussed offline, will leave to a follow up PR where we could created an `AuthActivityStarterHost` and save/clear it inside the `PaymentSheetViewModel` when `PaymentSheetActivity` is created/destoryed